chaincfg/params.go

```diff
@@ -415,6 +440,17 @@ var RegressionNetParams = Params{
 				time.Time{}, // Never expires
 			),
 		},
+		DeploymentTestDummyMinActivation: {
+			BitNumber:                 22,
+			CustomActivationThreshold: 72,  // Only needs 50% hash rate.
+			MinActivationHeight:       600, // Can only activate after height 600.
```

Comment:
Not sure I understand this correctly, but does this mean Taproot would only activate after 600 blocks on regtest? That would require us to mine quite a few blocks during integration tests and similar tasks.

Reply:
For tapproot, we'd set a height of zero so it can activate at anytime. This new deployment was added so I could test the "speedy trial" specific additions like the min activation height. 

This PR removed the old taproot deployment, as it can actually cause a panic in certain cases, as it needs to be defined for all network types (was only defined for segnet before).